### PR TITLE
Updated gen_defaults to use new JSON output from the Material token database.

### PR DESCRIPTION
**Pull request by darrenaustin**
The internal token extraction script has been updated and now spits out separate JSON files for each component. This PR updates the `gen_defaults` script to handle this.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: dev/tools/gen_defaults/data/color_dark.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "md.sys.color.primary": "md.ref.palette.primary80",
+  "md.sys.color.primary-container": "md.ref.palette.primary30",
+  "md.sys.color.secondary": "md.ref.palette.secondary80",
+  "md.sys.color.secondary-container": "md.ref.palette.secondary30",
+  "md.sys.color.shadow": "md.ref.palette.neutral0",
+  "md.sys.color.surface": "md.ref.palette.neutral10",
+  "md.sys.color.surface-variant": "md.ref.palette.neutral-variant30",
+  "md.sys.color.tertiary": "md.ref.palette.tertiary80",
+  "md.sys.color.tertiary-container": "md.ref.palette.tertiary30"
+}
```
Comment (rami-a) [inline dev/tools/gen_defaults/data/color_dark.json:31]: Nit: Newline at end of all these json files?

File: dev/tools/gen_defaults/data/shape.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  
+    "md.sys.shape.corner.extra-small.top": {
+      "family": "SHAPE_FAMILY_ROUNDED_CORNERS",
+      "topLeft": 4.0,
+      "topRight": 4.0,
+      "bottomLeft": 0.0,
+      "bottomRight": 0.0
+    },
+  
+
```
Comment (rami-a) [inline dev/tools/gen_defaults/data/shape.json:40]: Extra new lines between each entry in this file

**Review by rami-a — APPROVED** [review]
LGTM

**Review by HansMuller — APPROVED** [review]
LGTM

File: dev/tools/gen_defaults/data/color_dark.json (hunk 1 above)
Comment (darrenaustin) [reply dev/tools/gen_defaults/data/color_dark.json:31]: Caused by the token extractor script. I'll see what I can do next time I am updating the JSON files.

Comment (darrenaustin) [reply dev/tools/gen_defaults/data/color_dark.json:31]: Fixed.

File: dev/tools/gen_defaults/data/shape.json (hunk 2 above)
Comment (darrenaustin) [reply dev/tools/gen_defaults/data/shape.json:40]: Fixed by having the extractor script run clang format over the files.

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Pull request merged**